src/consensus/aft/raft.h

```diff
@@ -116,7 +116,7 @@ namespace aft
     // the current view. This signature is the first thing they may commit, as
     // they cannot confirm commit of anything from a previous view (Raft paper
     // section 5.4.2). This bool is true from the point this node becomes
-    // primary, until it is explicitly reset by a call to get_signable_txid()
+    // primary, until it sees a committable entry
```

Comment:
This is a drive-by, tangentially related change, so warrants some scrutiny. As this comment explains, we added a bool to track when this node has recently become primary and would like to emit its first signature in the new term. We previously reset this inside `get_signable_txid()`, with a few scary comments about assuming that the calling path proceeded to actually emit a signature. I think its simpler and safer to reset the bool later, when the emitted signature reaches Raft in `replicate()`. In practice this means "slightly later in exactly the same execution stack", and in theory introduces "we might produce multiple start-of-term signatures if things get weirdly delayed" and "we might be automagically retry if a first signing attempt fails". Leaning heavily on the existing tests to check that this is safe.

Reply:
Someone other than me should review this code change but from a consensus perspective, I have a strong preference for might produce multiple sigs over might not emit any sigs at the start of a term

Reply:
Unless this change is somehow necessary to remove the claims, I would like to see it split and some tests added to check how many extra signatures, if any, it results in.

Reply:
Sure, I've reverted this for now and will do it in a separate PR.